src/components/Wizard/CreateVmWizard/BasicSettingsTab.js

```diff
@@ -265,24 +266,40 @@ const asValueObject = (value, validation) => ({
   validation,
 });
 
-// TODO: To avoid race conditions, it would be better to retrieve basicSettings at the time of its actual useto align behavior with the setState()
+// TODO: To avoid race conditions as the basicSettings tab is bound to the CreateVmWizard state at the render-time, it would
+//  be better to retrieve basicSettings at the time of its actual use - to align behavior with the setState().
+// The onChange() bellow should be called with just the diff, not whole copy of the stepData.
 const publish = ({ basicSettings, templates, onChange, dataVolumes }, value, target, formValid, formFields) => {
-  const newBasicSettings = {
-    ...basicSettings,
-    [target]: value,
-  };
+  let newBasicSettings;
+  if (target === BATCH_CHANGES_KEY) {
+    // the "value" is an array of pairs { value, target }
+    const difference = value.value.reduce((map, obj) => {
+      map[obj.target] = { value: obj.value, validation: undefined };
```

Comment:
you should not set validation to undefined. See my comment below

Reply:
done